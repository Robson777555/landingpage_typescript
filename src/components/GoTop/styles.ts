import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;