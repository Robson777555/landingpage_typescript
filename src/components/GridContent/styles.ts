// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;