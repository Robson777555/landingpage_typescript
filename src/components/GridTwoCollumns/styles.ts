// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
