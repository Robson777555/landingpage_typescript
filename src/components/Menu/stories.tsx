import { Meta, StoryObj } from "@storybook/react";
import { Menu } from "./Menu";
import linksMock from "../NavLinks/mock";

interface LogoData {
  text: string;
  link: string;
  srcImg?: string;
}

export default {
  title: "Components/Menu",
  component: Menu,
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    links: {
      control: {
        type: "object",
      },
    },
    logoData: {
      control: {
        type: "object",
      },
    },
  },
  args: {
    links: linksMock,
    logoData: {
      text: "Logo",
      link: "#target",
      srcImg: "",
    } as LogoData,
  },
} as Meta<typeof Menu>;

type Story = StoryObj<typeof Menu>;

export const Desktop: Story = {};

export const Mobile: Story = {
  parameters: {
    viewport: {
      defaultViewport: "mobile1",
    },
  },
};
