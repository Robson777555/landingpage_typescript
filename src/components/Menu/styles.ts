import styled, { css } from 'styled-components';

interface ContainerProps {
  $isVisible?: boolean;
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, $isVisible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 100;
    background: ${theme.colors.white};
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
    transition: all 0.3s ease;
    height: ${$isVisible ? '100vh' : '60px'};
    overflow: ${$isVisible ? 'auto' : 'hidden'};

    @media (min-width: 769px) {
      position: static;
      height: auto;
      overflow: visible;
      box-shadow: none;
    }
  `}
`;

export const MenuContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    padding: ${theme.spacings.medium} 0;
    gap: ${theme.spacings.large};

    @media (min-width: 769px) {
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      padding: ${theme.spacings.small} 0;
      gap: ${theme.spacings.medium};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;
    position: fixed;
    top: ${theme.spacings.small};
    right: ${theme.spacings.small};
    z-index: 101;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    cursor: pointer;
    transition: transform 0.3s ease;

    &:hover {
      transform: scale(1.05);
    }

    @media (max-width: 768px) {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    svg {
      width: 2.4rem;
      height: 2.4rem;
    }
  `}
`;