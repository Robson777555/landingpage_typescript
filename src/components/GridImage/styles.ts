// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
