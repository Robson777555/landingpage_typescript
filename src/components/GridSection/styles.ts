import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
