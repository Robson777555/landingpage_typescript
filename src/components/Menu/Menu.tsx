'use client';
import React, { useState, useEffect } from 'react';
import * as Styled from './styles';
import { SectionContainer } from '../SectionContainer/SectionContainer';
import { LogoLink } from '../LogoLink/LogoLink';
import { NavLinks } from '../NavLinks/NavLinks';
import { Menu as MenuIcon } from '@styled-icons/material-outlined/Menu';
import { Close as CloseIcon } from '@styled-icons/material-outlined/Close';

interface LinkProps {
  children: string;
  link: string;
  newTab?: boolean;
}

interface LogoDataProps {
  text: string;
  link: string;
  srcImg?: string;
}

interface MenuProps {
  links?: LinkProps[];
  logoData: LogoDataProps;
}

export const Menu = ({ links = [], logoData }: MenuProps) => {
  const [isVisible, setIsVisible] = useState(false);
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const checkMobile = () => {
      setIsMobile(window.innerWidth <= 768);
    };

    // Verificação inicial e listener para redimensionamento
    checkMobile();
    window.addEventListener('resize', checkMobile);

    return () => {
      window.removeEventListener('resize', checkMobile);
    };
  }, []);

  const handleLinkClick = () => {
    if (isMobile) {
      setIsVisible(false);
    }
  };

  return (
    <>
      <Styled.Container $isVisible={isVisible}>
        <SectionContainer>
          <Styled.MenuContent>
            <LogoLink {...logoData} />
            <NavLinks links={links} onLinkClick={handleLinkClick} />
          </Styled.MenuContent>
        </SectionContainer>
      </Styled.Container>

      {isMobile && (
        <Styled.Button
          onClick={() => setIsVisible((prev) => !prev)}
          aria-label={isVisible ? 'Fechar menu' : 'Abrir menu'}
          aria-expanded={isVisible}
        >
          {isVisible ? (
            <CloseIcon aria-hidden="true" />
          ) : (
            <MenuIcon aria-hidden="true" />
          )}
        </Styled.Button>
      )}
    </>
  );
};

export default Menu;