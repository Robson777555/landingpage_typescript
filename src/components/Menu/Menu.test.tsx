import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Menu } from './Menu';
import { theme } from '../../styles/theme';
import linksMock from '../NavLinks/mock';
import { describe, it, expect, beforeEach } from 'vitest';

interface LogoData {
  text: string;
  link: string;
}

const logoData: LogoData = {
  text: 'Logo',
  link: '#target',
};

describe('<Menu />', () => {
  const originalInnerWidth = window.innerWidth;

  beforeEach(() => {
    // Reset window size before each test
    window.innerWidth = originalInnerWidth;
  });

  const setupMobile = (isMobile: boolean) => {
    window.innerWidth = isMobile ? 500 : 1024;
    fireEvent(window, new Event('resize'));
  };

  it('should render logo and navigation on desktop', () => {
    setupMobile(false);

    render(
      <ThemeProvider theme={theme}>
        <Menu links={linksMock} logoData={logoData} />
      </ThemeProvider>
    );

    expect(screen.getByText('Logo')).toBeInTheDocument();
    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(screen.queryByLabelText('Abrir menu')).toBeNull();
    expect(screen.queryByLabelText('Fechar menu')).toBeNull();
  });

  it('should toggle menu on mobile', () => {
    setupMobile(true);

    render(
      <ThemeProvider theme={theme}>
        <Menu links={linksMock} logoData={logoData} />
      </ThemeProvider>
    );

    const button = screen.getByLabelText('Abrir menu');
    fireEvent.click(button);
    expect(screen.getByLabelText('Fechar menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByLabelText('Abrir menu')).toBeInTheDocument();
  });

  it('should close menu when clicking a link on mobile', () => {
    setupMobile(true);

    render(
      <ThemeProvider theme={theme}>
        <Menu links={linksMock} logoData={logoData} />
      </ThemeProvider>
    );

    fireEvent.click(screen.getByLabelText('Abrir menu'));
    fireEvent.click(screen.getByText(linksMock[0].children));
    expect(screen.getByLabelText('Abrir menu')).toBeInTheDocument();
  });
  it(' Snapshpt page ', () => {
    const renderMenu = () => {
      return render(
        <ThemeProvider theme={theme}>
          <Menu links={linksMock} logoData={logoData} />
        </ThemeProvider>
      );
    };
    const { container } = renderMenu();
    expect(container.firstChild).toMatchSnapshot();
  });
});