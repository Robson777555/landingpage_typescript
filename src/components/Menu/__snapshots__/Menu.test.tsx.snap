// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Menu /> >  Snapshpt page  1`] = `
<div
  class="sc-bXdtCk dqbQEq"
>
  <div
    class="sc-gwsNht cxvrue"
    data-testid="section-container"
  >
    <div
      class="sc-hKDTPf dPoFM"
    >
      <h1
        class="sc-jMpmlX eITWUX"
        data-testid="heading"
      >
        <a
          class="sc-dYOLZc etiJDl"
          href="#target"
          target="_self"
        >
          <span>
            Logo
          </span>
        </a>
      </h1>
      <nav
        aria-label="Main menu"
        class="sc-jZhnRx pXseN"
        role="navigation"
      >
        <a
          class="sc-hoLldG brFPWH"
          href="#target1"
          target="_self"
        >
          Servi√ßos
        </a>
        <a
          class="sc-hoLldG brFPWH"
          href="#target2"
          target="_self"
        >
          Tecnologias usadas
        </a>
        <a
          class="sc-hoLldG brFPWH"
          href="#target3"
          target="_self"
        >
          Tecnologias essenciais
        </a>
      </nav>
    </div>
  </div>
</div>
`;
